src/bootstrap/src/core/build_steps/synthetic_targets.rs

```diff
@@ -59,6 +59,7 @@ fn create_synthetic_target(
     let mut cmd = Command::new(builder.rustc(compiler));
     cmd.arg("--target").arg(base.rustc_target_arg());
     cmd.args(["-Zunstable-options", "--print", "target-spec-json"]);
+    cmd.env("RUSTC_BOOTSTRAP", "1");
```

Comment:
is this still needed with the other changes? If so, please leave a comment explaining that this is for custom targets

Reply:
This isn't for custom targets. When `rust.channel` is set to either beta or stable, we can't use nightly features on bootstrap without `RUSTC_BOOTSTRAP`. I will add an explanatory comment.